### DISCUSSION TITLE
TARDIS addon integration

### DISCUSSION THREAD

**Discussion by DalekCraft2** [eccentricdevotion/TARDIS#514] [Q&A] [answered]
I am wondering about whether I may integrate TARDIS-Dynmap and TARDISChunkGenerator into the base plugin. The first is very simple, and the second is necessary for the base plugin to function.

**Comment by Chew** [comment]
If you're dependant on them you can put it in your plugin.yml, not sure why you would need to just copy the code since they are plugins. 
Anyway, the answer is technically/legally yes because of GPL and all that. Should give credit though.

**Reply by DalekCraft2** [reply]
Well, I am doing this for a pull request.

**Reply by Chew** [reply]
Oh I misread, I'm sure they're separate for a reason, what that reason is we'll find out 😂

**Comment by eccentricdevotion** [comment, marked as answer]
Not possible I believe.
You need to consider load order and dependencies.
TARDISChunkGenerator is a world generator so has `load: STARTUP` in plugin.yml so loads before any worlds are loaded.
Dynmap-TARDIS has `load: POSTWORLD` in plugin.yml, and also a dependency on both the Dynmap plugin itself and the TARDIS plugin so needs to load after both plugins, and after worlds have loaded.

**Reply by eccentricdevotion** [reply]
You could however, combine Dynmap-TARDIS and the TARDIS plugin.

**Reply by DalekCraft2** [reply]
Would it be possible to move the non-chunk-generation files, such as disguises, to the main plugin? I want to integrate as much as possible.

**Reply by eccentricdevotion** [reply]
The reason for most of the code in TCG being there is because it uses Minecraft and CraftBukkt internals. It's easier to keep that code separate so that it is quicker to update and troubleshoot when new Minecraft versions are released.

**Reply by eccentricdevotion** [reply]
I really don't see any benefit in combining plugins...

**Reply by DalekCraft2** [reply]
I do not see the benefit in keeping them separate if they require each other to function.
